src/argilla/client/feedback/metrics/annotator_metrics.py

```diff
@@ -0,0 +1,373 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""This module contains metrics to compare Annotator's suggestions vs responses. """
+
+from abc import ABC, abstractmethod
+from collections import defaultdict
+from typing import TYPE_CHECKING, Any, Dict, List, Tuple, Union
+
+import numpy as np
+from pydantic import BaseModel
+
+from argilla.client.feedback.metrics.utils import get_responses_per_user
+from argilla.client.feedback.schemas import (
+    LabelQuestion,
+    MultiLabelQuestion,
+    RankingQuestion,
+    RatingQuestion,
+    TextQuestion,
+)
+from argilla.utils.dependency import requires_dependencies
+
+if TYPE_CHECKING:
+    from argilla.client.feedback.dataset import FeedbackDataset
+
+import pandas as pd
+from sklearn.metrics import (
+    accuracy_score,
+    confusion_matrix,
+    f1_score,
+    multilabel_confusion_matrix,
+    precision_score,
+    recall_score,
+)
+
+
+class AnnotatorMetricResult(BaseModel):
+    """Container for the result of an annotator metric.
+
+    It contains two fields, `metric_name` and `result` with the value of the metric.
+    """
+
+    metric_name: str
+    result: Union[float, Dict[str, float], pd.DataFrame, List[pd.DataFrame]]
+
+    class Config:
+        arbitrary_types_allowed = True
+
+
+class AnnotatorMetric:
+    def __init__(self, dataset: "FeedbackDataset", question_name: str) -> None:
+        """Initializes a `AnnotatorMetric` object to compute metrics on
+        a `FeedbackDataset` for a given question at the annotator level.
+
+        Args:
+            dataset: FeedbackDataset to compute the metrics.
+            question_name: Name of the question for which we want to analyse the analyze
+                the annotator responses vs the suggestions given.
+
+        Raises:
+            NotImplementedError: If the question type is not supported.
+        """
+        self._dataset = dataset
+        self._question_name = question_name
+        self._question_type = type(self._dataset.question_by_name(question_name))
+        if self._question_type in (MultiLabelQuestion, RankingQuestion):
+            raise NotImplementedError(f"No metrics are defined currently for {self._question_type.__name__}")
```

Comment:
perhaps we can pro-actively create an issue and redirect users there to start a discussion?